pages/[username].js

```diff
@@ -23,6 +23,16 @@ export async function getServerSideProps(context) {
   });
 
   const { status, profile } = await getUserApi(req, res, username);
+
+  if (status === 404) {
+    return {
+      redirect: {
+        destination: `/404`,
+        permanent: false,
+      },
+    };
+  }
```

Comment:
This is not needed, as it will go to the `search` page from the code below if no user is found, which I think is better user experience